dubbo-common/src/main/java/org/apache/dubbo/common/utils/ClassHelper.java

```diff
@@ -265,10 +265,14 @@ public static boolean isPrimitive(Class<?> type) {
     }
 
     public static Object convertPrimitive(Class<?> type, String value) {
-        if (type == char.class || type == Character.class) {
-            return value.length() > 0 ? value.charAt(0) : '\0';
+        if (value == null) {
+            return null;
+        } else if (type == char.class || type == Character.class) {
+            return value.length() > 0 ? value.charAt(0) : null;
         } else if (type == boolean.class || type == Boolean.class) {
             return Boolean.valueOf(value);
+        } else if (StringUtils.isRealBlank(value)) {
```

Comment:
Hi, 

I think we don't need to call `isRealBlank`, because there is too much to check. If we really want to check, we need to make sure the input `value` is numeric. I have left comment on #3093